personalities/sctfe/storage.go

```diff
@@ -78,3 +87,65 @@ func (cts *CTStorage) AddIssuerChain(ctx context.Context, chain []*x509.Certific
 	}
 	return nil
 }
+
+// cachedIssuerStorage wraps an IssuerStorage, and keeps a copy the sha256 of certs it contains.
+//
+// This is intended to make querying faster. It does not keep a copy of the certs, only sha256.
+// Only up to N keys will be stored locally.
+// TODO(phboneff): add monitoring for the number of keys
+type cachedIssuerStorage struct {
+	m map[string]bool
+	N int // maximum number of entries allowed in m
+	s IssuerStorage
+}
+
+// Exists checks if the key exists in the local cache, if not checks in the underlying storage.
+// If it finds it there, caches the key locally.
+func (c cachedIssuerStorage) Exists(ctx context.Context, key []byte) (bool, error) {
+	_, ok := c.m[string(key)]
+	if ok {
```

Comment:
the comment at the top suggests that map keys are 64byte - is `key` here already hex encoded?

If not, then might be worth updating the comment, if it is, then L107 below will re-hexencode it.